cob_bms_driver/src/power_state_aggregator.py

```diff
@@ -63,11 +71,26 @@ def calculate_power_consumption(self):
 
     def calculate_relative_remaining_capacity(self):
         if self.full_charge_capacity != None and self.remaining_capacity != None:
-            return round(100.0*(self.remaining_capacity/self.full_charge_capacity), 3)
+            if self.voltage <= self.threshold_voltage_critical and self.current > -self.steady_current:
+                self.relative_remaining_capacity = 5 # % of full capacity
+            elif self.voltage <= self.threshold_voltage_error and self.current > -self.steady_current:
+                self.relative_remaining_capacity = 10 # % of full capacity
+            elif self.voltage <= self.threshold_voltage_warn and self.current > -self.steady_current:
+                self.relative_remaining_capacity = 15 # % of full capacity
+            elif self.voltage <= self.threshold_voltage_mid and self.current > -self.steady_current:
+                self.relative_remaining_capacity = 50 # % of full capacity
+            else: #self.voltage <= self.threshold_voltage_full and self.current < self.steady_current:
+                self.relative_remaining_capacity = 100 # % of full capacity
+
+            return self.relative_remaining_capacity
+
         else:
             rospy.logwarn("cannot calculate relative remaining capacity. full_charge_capacity=%d, remaining_capacity=%d" % (self.full_charge_capacity, self.remaining_capacity))
             return 0.0
 
+        return self.relative_remaining_capacity
```

Comment:
If so, this will fail as this is member is not initialized.
However, three is a `return` in both the `if` and the `else` branch, so we will not get to this part...